splitgpg2/__init__.py

```diff
@@ -819,44 +838,51 @@ def escape_D(data):
     # we send the reserialized form this should be safe.
 
     @classmethod
-    def parse_sexpr(klass, untrusted_arg):
+    def parse_sexpr(klass, untrusted_arg: bytes):
+        if type(untrusted_arg) is not bytes:
+            raise TypeError("invalid type in parse_sexpr")
         if len(untrusted_arg) == 0:
             raise ValueError("no sexpr")
+        if untrusted_arg[0] != ord('('):
+            raise ValueError("sexpr must start with (")
```

Comment:
Added a check that looks nicer to me. Please keep not directly related changes in separate commits.